### DISCUSSION TITLE
Cosmos Operator and Bare Metal

### DISCUSSION THREAD

**Discussion by seb3point0** [strangelove-ventures/cosmos-operator#343] [General]
Hey guys - appreciate the work you're doing. 

We're thinking of running Cosmos Operator in a bare metal setup so multiple chains can run on the same box. The goal is to optimize for cost on smaller chains, not requiring high resources or testnets. 

K8S Master: Inexpensive cloud server
K8s Worker 1: Bare Metal server
- Pod: Mainnet 1
- Pod: Mainnet 2
- Pod: Mainnet 3
K8s Worker 2: Bare Metal server
- Pod: Testnet 1
- Pod: Testnet 2
- Pod: Testnet 3

Would this setup work with Cosmos Operator? What's important not to overlook in such a setup?

**Comment by DavidNix** [comment]
> Would this setup work with Cosmos Operator?

Unfortunately, this is impossible for us to predict. There's numerous ways to configure a bare metal setup. It is complex and requires a lot of skill.

> What's important not to overlook in such a setup?

We've started moving some of our workloads to a bare metal cluster, primarily to save on cloud costs.

What has worked for us so far is:
- https://cilium.io for networking
- Kubeadm
- https://github.com/topolvm/topolvm for volume management

But there will be surprises. For example, we had issues downloading snapshots that related to CoreDNS and IPv6. 

We may be able to offer consulting to help you set up a bare metal cluster specifically for your use case. If interested, please let us know and we can get you in touch with our biz dev team.

@agouin has lead our bare metal migration. He may want to chime in with more details.
